lib/puppet/parser/scope.rb

```diff
@@ -18,7 +18,6 @@ class Puppet::Parser::Scope
   extend Forwardable
   include Puppet::Util::MethodHelper
 
-  include Puppet::Resource::TypeCollectionHelper
```

Comment:
The `Puppet::Resource::TypeCollectionHelper` is also included in `Puppet::Parser::Resource` and `Puppet::Pops::Parser`. Suggest removing those includes too, and then also remove the actual module.